File: src/main/java/ru/vmsystems/template/interfaces/dto/MaterialDto.java
package ru.vmsystems.template.interfaces.dto;

import lombok.Data;

import java.util.Objects;

@Data
public class MaterialDto {
    private Long id;
    private Integer depth;
    private Double length;
    private Double width;
    private String price;
    private String desc;
    private String name;
    private MaterialColorDto color;
    private MaterialTypeDto type;

    public void setName() {
        name = String.join(" ", type.getName(), color.getName(), String.valueOf(depth), " мм");
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        MaterialDto that = (MaterialDto) o;
        return Objects.equals(id, that.id) &&
                Objects.equals(depth, that.depth) &&
                Objects.equals(length, that.length) &&
                Objects.equals(width, that.width) &&
                Objects.equals(price, that.price) &&
                Objects.equals(desc, that.desc) &&
                Objects.equals(name, that.name) &&
                Objects.equals(color, that.color) &&
                Objects.equals(type, that.type);
    }

    @Override
    public int hashCode() {

        return Objects.hash(id, depth, length, width, price, desc, name, color, type);
    }
}

File: src/main/java/ru/vmsystems/glasscloud/domain/material/MaterialDto.kt
package ru.vmsystems.glasscloud.domain.material

import java.math.BigDecimal
import java.util.*

data class MaterialDto(
        var id: UUID?,
        val deleted: Boolean = false,
        val depth: Float,
        val length: Float,
        val width: Float,
        val price: BigDecimal,
        val description: String?,
        val color: MaterialColorDto,
        val type: MaterialTypeDto,
        val companyId: UUID?
)

data class MaterialColorDto(
        var id: UUID?,
        val name: String,
        val deleted: Boolean = false,
        val companyId: UUID?
) {
//    constructor(id: String, name: String, deleted: Boolean, companyId: String) : this(
//            id = UUID.fromString(id),
//            name = name,
//            deleted = deleted,
//            companyId = UUID.fromString(companyId)
//    )
}

data class MaterialTypeDto(
        var id: UUID?,
        val name: String,
        val deleted: Boolean = false,
        val companyId: UUID?
)